internal/nodejs-action/generate-layout.sh

```diff
@@ -2,7 +2,35 @@
 
 set -euo pipefail
 
-package_id=$(echo "${PACK_JSON}" | jq -r '.[0].id')
+# The subject name is an npm package url (purl).
+# https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst
+#
+# The npm package's scope is considered a purl "namespace" and not part of the
+# package name. So the subject will take the form of:
+#
+# With scope:
+#   pkg:npm/<scope>/<name>@<version>
+#
+# Without scope:
+#   pkg:npm/<name>@<version>
+raw_package_scope=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f1)
+raw_package_name=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f2)
+if [ "${raw_package_name}" == "" ]; then
+    raw_package_name="${raw_package_scope}"
+    raw_package_scope=""
+fi
+# package scope (namespace) is URL(percent) encoded.
+package_scope=$(echo "\"${raw_package_scope}\"" | jq -r '. | @uri')
+# package name is URL(percent) encoded.
+package_name=$(echo "\"${raw_package_name}\"" | jq -r '. | @uri')
+# version is URL(percent) encoded.
+package_version=$(echo "${PACK_JSON}" | jq -r '.[0].version | @uri')
```

Comment:
maybe add a comment that version may be a commit, or a string provided by user

Reply:
Done.